subscriptions-manager/src/main/java/io/arlas/subscriptions/dao/MongoUserSubscriptionDAOImpl.java

```diff
@@ -75,20 +77,28 @@ public MongoUserSubscriptionDAOImpl(MongoDBConnection mongoDBConnection, MongoDB
                 filters.add(eq("deleted", Boolean.FALSE));
             if (active != null)
                 filters.add(eq("active", active));
+            if (started != null)
+                filters.add(started ? lte("starts_at", System.currentTimeMillis() / 1000l) : gt("started", System.currentTimeMillis() / 1000l));
+            if (createdByAdmin != null)
+                filters.add(eq("created_by_admin", createdByAdmin));
             if (expired != null)
                 filters.add(expired ? lte("expires_at", System.currentTimeMillis() / 1000l) : gt("expires_at", System.currentTimeMillis() / 1000l));
             if (before != null)
                 filters.add(lte("created_at", before));
+            if (after != null)
+                filters.add(gte("created_at", after));
 
-            Document aggResult = this.mongoCollectionDoc.aggregate(
-                    Arrays.asList(
-                            match(and(filters)),
-                            facet(
-                                    new Facet("subList", skip(size * (page - 1)), limit(size)),
-                                    new Facet("totalCount", count())
-                            )
-                    )
-            ).first();
+            List<Bson> aggregate = new ArrayList<>();
+            aggregate.add(sort(Sorts.descending("created_at")));
+            if (!filters.isEmpty()) {
+                aggregate.add(match(and(filters)));
+            }
+            aggregate.add(facet(
```

Comment:
I wonder how it behaves when the filters are empty. The sort alone is enough to get a list of result? Have you tested?

Reply:
I did. If you apply no filter at all, by default the controller set deleted=true by this is ignored then, and the `filters`is empty. This is part of the integration tests.